File: backend/app/schemas/resume.py
"""Resume Builder Schemas"""
from datetime import datetime
from typing import Optional, List, Dict, Any
from pydantic import BaseModel, Field


class ResumeContentSection(BaseModel):
    """Resume content section"""
    type: str  # summary, experience, education, skills, etc.
    content: Dict[str, Any]


class ResumeCreate(BaseModel):
    """Create resume"""
    title: str = Field(..., max_length=200)
    template_id: str = "professional"
    content: Dict[str, Any] = {}


class ResumeUpdate(BaseModel):
    """Update resume"""
    title: Optional[str] = Field(None, max_length=200)
    template_id: Optional[str] = None
    content: Optional[Dict[str, Any]] = None
    is_primary: Optional[bool] = None


class ResumeResponse(BaseModel):
    """Resume response"""
    id: int
    user_id: int
    title: str
    template_id: str
    content: Dict[str, Any]
    ats_score: Optional[float]
    keywords: List[str]
    is_primary: bool
    version: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class ResumeExportRequest(BaseModel):
    """Export resume"""
    format: str = Field(..., pattern="^(pdf|docx|txt)$")


class ResumeExportResponse(BaseModel):
    """Export response"""
    id: int
    resume_id: int
    format: str
    file_url: str
    file_size: int
    created_at: datetime

    class Config:
        from_attributes = True


class ATSOptimizationRequest(BaseModel):
    """ATS optimization request"""
    job_id: Optional[int] = None
    job_description: Optional[str] = None


class ATSOptimizationResponse(BaseModel):
    """ATS optimization response"""
    id: int
    resume_id: int
    job_id: Optional[int]
    score: float
    suggestions: List[Dict[str, Any]]
    missing_keywords: List[str]
    formatting_issues: List[Dict[str, Any]]
    created_at: datetime

    class Config:
        from_attributes = True


class AIContentSuggestion(BaseModel):
    """AI content suggestion"""
    section: str
    original: str
    suggested: str
    reason: str
    impact: str  # high, medium, low

File: frontend/src/types/resume.ts
/**
 * Resume Builder Types
 */

export interface Resume {
  id: number;
  user_id: number;
  title: string;
  template_id: string;
  content: ResumeContent;
  ats_score?: number;
  keywords: string[];
  is_primary: boolean;
  version: number;
  created_at: string;
  updated_at: string;
}

export interface ResumeContent {
  summary?: string;
  experience?: WorkExperience[];
  education?: Education[];
  skills?: string[];
  certifications?: Certification[];
  projects?: Project[];
}

export interface WorkExperience {
  company: string;
  position: string;
  location?: string;
  start_date: string;
  end_date?: string;
  current: boolean;
  description: string;
  achievements: string[];
}

export interface Education {
  institution: string;
  degree: string;
  field: string;
  start_date: string;
  end_date?: string;
  gpa?: number;
}

export interface Certification {
  name: string;
  issuer: string;
  date: string;
  credential_id?: string;
  url?: string;
}

export interface Project {
  name: string;
  description: string;
  technologies: string[];
  url?: string;
}

export interface AIContentSuggestion {
  section: string;
  original: string;
  suggested: string;
  reason: string;
  impact: 'high' | 'medium' | 'low';
}

export interface ATSOptimization {
  id: number;
  resume_id: number;
  job_id?: number;
  score: number;
  suggestions: ATSSuggestion[];
  missing_keywords: string[];
  formatting_issues: FormattingIssue[];
  created_at: string;
}

export interface ATSSuggestion {
  type: string;
  message: string;
  priority: string;
}

export interface FormattingIssue {
  issue: string;
  location: string;
  fix: string;
}

export interface ResumeTemplate {
  id: string;
  name: string;
  description: string;
  preview_url: string;
}
